Fix logging

kentsanggds opened the pull request:
## What

Logs were not set up correctly, this PR should fix the logging as well as bring the stack more in line with the DGU dev stack.

## How to review

As there are still lots of issues to resolve in the structure of the project and there needs to be a bit more time to think about how much we can deviate from the ofkn base repo, I think it would be best to concentrate on just bringing up the ckan stack, navigating to http://localhost:5555 and then check that logs are appearing in the host machine.

The README should help with the setup, if they are insufficient they can be updated, and the helper scripts mentioned should work.

## Reference 

https://trello.com/c/YKMrKVsP/1482-5-fix-the-logging-configuration-on-docker-ckan%F0%9F%8D%90

kentsanggds committed:
Update dockerfile.dev to create CKAN dev stack

kentsanggds committed:
Change port to 5555

- so that we can run vagrant stack alongside the docker stack to compare differences

kentsanggds committed:
Update ckan-dev to use own Dockerfile and setup harvest process

Also add helper scripts to rebuild Docker images for ckan-base and ckan-dev

kentsanggds committed:
Add script to reset docker

kentsanggds committed:
Change container name db to ckan_db

kentsanggds committed:
Add alpine-geos image

This image was built because the update to the alpine geo dependency broke the installation of geos preventing the installation of shapely used by ckanext-spatial.

The Dockerfile uses a version of geos which works with the alpine docker image.

kentsanggds committed:
Remove harvest conf as not needed as part of core ckan image

kentsanggds committed:
Update base Dockerfile

to install long dependencies
and indicate where geos can be installed if the geos dependency gets fixed on alpine

kentsanggds committed:
Update dev Dockerfile

install netcat to check for running service and create the ckan log folder

kentsanggds committed:
Update gather and harvest log paths

kentsanggds committed:
Update docker-compose to remove datapusher and setup docker volume for ckan logs on host machine

kentsanggds committed:
Allow execute shell scripts

kentsanggds committed:
Update README with information about new shell scripts and how to run docker ckan

kentsanggds committed:
Add example production.ini

kentsanggds committed:
Ignore logs

thomasleese commented on README.md:
```diff
@@ -54,14 +54,28 @@ To develop local extensions use the `docker-compose.dev.yml` file:
+docker exec -it ckan_db psql -U ckan
+
+export CKAN_INI=/srv/app/production.ini
+paster --plugin=ckan -c /srv/app/production.ini
```
It might be good to put this in backticks so it's rendered as code.

kentsanggds replied:
:+1:

thomasleese commented on alpine-geos/Dockerfile:
```diff
@@ -0,0 +1,13 @@
+# install geos 3.8.0 as geos 3.7.2-r1 in alpine repository does not install correctly
+RUN wget http://download.osgeo.org/geos/geos-3.8.0.tar.bz2
+RUN tar xf geos-3.8.0.tar.bz2
+RUN (cd geos-3.8.0 && ./configure && make && make install)
```
```suggestion
RUN cd geos-3.8.0 && ./configure && make && make install
```

Does this need brackets?

kentsanggds replied:
Probably not now, will change

thomasleese commented on ckan-dev/2.7/Dockerfile:
```diff
@@ -1,19 +1,25 @@
-FROM openknowledge/ckan-base:2.7
+# FROM openknowledge/ckan-base:2.7
+FROM alphagov/ckan-base:2.7
```
Does this mean we need to push `ckan-base` to our Dockerhub profile?

kentsanggds replied:
For now, we have to rebuild everything, as changes to the base happen fairly frequently, when things stabilise I could push something up to make the process a bit faster

thomasleese replied:
Ah that makes sense. I was wondering if we were forced to use it because of the `alphagov/` prefix, but I see it can take it from a locally built image.

thomasleese commented on ckan-dev/2.7/Dockerfile:
```diff
@@ -1,19 +1,25 @@
 RUN mkdir $SRC_EXTENSIONS_DIR
 
+# create folder for logs
+RUN mkdir '/var/log/ckan'
```
```suggestion
RUN mkdir /var/log/ckan
```

kentsanggds replied:
👍

kentsanggds committed:
Change ckan_db to db

ckan_db was not found so revert it for now.

Also start server in docker-compose instead so that we don't have to rebuild ckan-dev.

thomasleese committed:
Add shebang to reset-docker.sh

thomasleese committed:
Add shebang to start-ckan.sh

kentsanggds committed:
WIP Move run command back to ckan-dev Dockerfile

kentsanggds committed:
Add reset docker volumes

kentsanggds committed:
Target docker-compose.dev.yml

kentsanggds committed:
Remove postgis installations as already installed

thomasleese commented:
I've managed to get this working on my machine now. I'll merge this now, we can work on the structure of the project and the future of CKAN Docker (and how to move it into GOV.UK Docker) in the future.
